### ISSUE TITLE
PacketEvents API support as an alternative to protocolLib

### ISSUE THREAD

**Issue by ADM8** [WeaponMechanics/MechanicsMain#337]
### Description

PacketEvents API by retrooper as an alternative option instead of protocolLib. Mainly related to efficiency reasons.

### Configuration

Nothing in configs of note, just detects PacketEvents API instead of protocolLib and boots up, or is an alternate fork of WM.

### Payment (optional)

- [x] I'm willing to pay for this feature

**Comment by CJCrafter** [comment]
I'll have to look into the API to check how much work this is. Either way, I do not want to maintain alternate forks given how often I have to update 1 fork already. The solution would definitely be a compatibility layer to detect whether to use ProtocolLib or PacketEvents

**Comment by CJCrafter** [comment]
I'm going to close this since for a few reasons:
- I doubt that PacketEvents is faster. They call the Bukkit Events API for every packet, which is something that Spigot themselves try to avoid doing because of lag issues
- Adding a compatibility layer to support both ProtocolLib and PacketEvents would introduce overhead that is not worth the effort
- The usage of ProtocolLib in WeaponMechanics is quite minimal

Basically, https://stackify.com/premature-optimization-evil/

**Comment by CJCrafter** [comment]
Reopening this due to PacketEvents being updated faster than ProtocolLib
